uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/zones/ZoneControllerMockMvcTests.java

```diff
@@ -159,17 +163,70 @@ void testGetZone() throws Exception {
+            result.getResponse().getStatus() == 200) {
+            return JsonUtils.readValue(result.getResponse().getContentAsString(), ZoneResponse.class);
+        } else {
+            return null;
+        }
+    }
+
+    private void checkGetZoneAndAssertError(MockHttpServletRequestBuilder identityZone, ResultMatcher expectedStatus,
+                                            String expected) throws Exception {
+        MvcResult result = mockMvc.perform(identityZone
+                                          .header("Authorization", "Bearer " + identityClientZonesReadToken))
+                                  .andExpect(expectedStatus).andReturn();
+        assertEquals(expected, result.getResponse().getContentAsString());
     }
 
     @Test
     void testGetZone_Notfound() throws Exception {
-        MvcResult result = mockMvc.perform(
-                   get("/zones").param("name", "random-name")
-                                .header("Authorization", "Bearer " + identityClientZonesReadToken))
-               .andExpect(status().isNotFound()).andReturn();
-        assertEquals(
-            "{\"message\", \"Zone[random-name] not found.\" }",
-            result.getResponse().getContentAsString());
+        checkGetZoneAndAssertError(get("/orchestrator/zones").param("name", "random-name"), status().isNotFound(),
+                                   "{\"message\", \"Zone[random-name] not found.\" }");
+    }
+
+    @Test
+    void testDeleteZone() throws Exception {
+        //TODO: remove this createIdentityZone method once the create orchestrator zone API is implemented and use it to create zone.
+        IdentityZone identityZone = createIdentityZone();
+        uaaEventListener.clearEvents();
+        processZoneAPI(delete("/orchestrator/zones"), identityZone.getName(), status().isAccepted());
+        checkGetZoneAndAssertError(get("/orchestrator/zones").param("name", identityZone.getName()),
+                                   status().isNotFound(),
+                                   "{\"message\", \"Zone[test-name] not found.\" }");
+
+        // Asserting delete event
+        assertThat(uaaEventListener.getEventCount(), is(1));
+        AbstractUaaEvent event = uaaEventListener.getLatestEvent();
+        assertThat(event, instanceOf(EntityDeletedEvent.class));
+        EntityDeletedEvent deletedEvent = (EntityDeletedEvent) event;
+        assertThat(deletedEvent.getDeleted(), instanceOf(IdentityZone.class));
+    }
+
+    @Test
+    void testDeleteZone_ZoneNotFound() throws Exception {
+        checkGetZoneAndAssertError(delete("/orchestrator/zones").param("name", "random-name"),
+                                   status().isNotFound(),
+                                   "{\"message\", \"Zone[random-name] not found.\" }");
+    }
+
+    @Test
+    void testUpdateZone_MethodNotImplemented() throws Exception {
+        checkGetZoneAndAssertError(put("/orchestrator/zones").contentType(APPLICATION_JSON).content(
```

Comment:
As checkGetZoneAndAssertError() method is used for all type of requests GET, DELETE, PUT we can rename it to some common name maybe performMockMvcCallAndAssertError().

Reply:
Done